scipy/sparse/csgraph/tests/test_graph_laplacian.py

```diff
@@ -34,26 +45,32 @@ def _check_symmetric_graph_laplacian(mat, normed):
     else:
         sp_mat = sparse.csr_matrix(mat)
 
-    laplacian = csgraph.laplacian(mat, normed=normed)
+    mat_copy = np.copy(mat)
+    sp_mat_copy = sparse.csr_matrix(sp_mat, copy=True)
+
     n_nodes = mat.shape[0]
-    if not normed:
-        assert_array_almost_equal(laplacian.sum(axis=0), np.zeros(n_nodes))
-    assert_array_almost_equal(laplacian.T, laplacian)
-    assert_array_almost_equal(
-        laplacian, csgraph.laplacian(sp_mat, normed=normed).toarray())
+    explicit_laplacian = _explicit_laplacian(mat, normed=normed)
+    laplacian = csgraph.laplacian(mat, normed=normed, copy=copy)
+    sp_laplacian = csgraph.laplacian(sp_mat, normed=normed,
+                                     copy=copy)
+
+    if copy:
+        assert_array_almost_equal(mat, mat_copy)
+        _assert_allclose_sparse(sp_mat, sp_mat_copy)
+    else:
+        if not (normed and (np.issubdtype(mat.dtype, np.signedinteger)
+                            or np.issubdtype(mat.dtype, np.uint))):
+            assert_array_almost_equal(laplacian, mat)
+            if sp_mat.format == 'coo':
+                _assert_allclose_sparse(sp_laplacian, sp_mat)
 
-    assert_array_almost_equal(laplacian,
-            _explicit_laplacian(mat, normed=normed))
+    assert_array_almost_equal(laplacian, sp_laplacian.toarray())
 
-
-def test_laplacian_value_error():
-    for t in int, float, complex:
-        for m in ([1, 1],
-                  [[[1]]],
-                  [[1, 2, 3], [4, 5, 6]],
-                  [[1, 2], [3, 4], [5, 5]]):
-            A = np.array(m, dtype=t)
-            assert_raises(ValueError, csgraph.laplacian, A)
+    for tested in [laplacian, sp_laplacian.toarray()]:
+        if not normed:
+            assert_array_almost_equal(tested.sum(axis=0), np.zeros(n_nodes))
+        assert_array_almost_equal(tested.T, tested)
```

Comment:
Sorry just seeing this, but since we are touching these we should use `assert_allclose` instead as `assert_array_almost_equal` is not recommended anymore.

Reply:
done